### ISSUE TITLE
support kotlin objects

### ISSUE THREAD

**Issue by adelnizamutdinov** [nsk-mironov/smuggler#10]
`Parcelable` implementation for an `object` is pretty trivial:

```kotlin
inline fun <reified T : Parcelable> creator(crossinline f: (Parcel) -> T) =
    object : Parcelable.Creator<T> {
      override fun createFromParcel(source: Parcel): T = f(source)
      override fun newArray(size: Int): Array<T?> = arrayOfNulls(size)
    }

object Welcome : Parcelable {
  @JvmField val CREATOR = creator { this }
  override fun describeContents(): Int = 0
  override fun writeToParcel(dest: Parcel?, flags: Int): Unit = Unit
}
```

what do you think?

**Comment by nsk-mironov** [comment]
Ok, will do. Should be quite easy to implement.

**Comment by nsk-mironov** [comment]
@adelnizamutdinov I've just published `0.12.4` update with kotlin objects support.

**Comment by adelnizamutdinov** [comment]
wow, that was fast : )
